CanHazFunny/CanHazFunny.Tests/JesterTests.cs

```diff
@@ -1,7 +1,67 @@
-using Xunit;
-
-namespace CanHazFunny.Tests;
-
-public class JesterTests
-{
-}
+using Xunit;
+using System;
+using Moq;
+
+namespace CanHazFunny.Tests;
+
+public class JesterTests
+{
+    [Fact]
+    public void TellJoke_JokeOutputNull_ThrowsException()
+    {
+        // Arrange
+        var moqJokeService = new Mock<IJokeService>();
+        
+        // Act and Assert
+        Assert.Throws<ArgumentNullException>(() => new Jester(moqJokeService.Object, null!));
+    }
+
+    [Fact]
+    public void TellJoke_JokeServiceNull_ThrowsException()
+    {
+        // Arrange
+        var moqJokeOutput = new Mock<IJokeOutput>();
+
+        // Act and Assert
+        Assert.Throws<ArgumentNullException>(() => new Jester(null!, moqJokeOutput.Object));
+    }
+
+
+    [Fact]
+    public void OutputJoke_WithoutChuckNorris_Successful()
+    {
+        // Arrange
+        var moqJokeService = new Mock<IJokeService>();
+        var moqJokeOutput = new Mock<IJokeOutput>();
+        string testJoke = "test joke";
+        moqJokeService.SetupSequence(x => x.GetJoke()).Returns(testJoke);
+        Jester jesterTest = new(moqJokeService.Object, moqJokeOutput.Object);
+        
+        // Act
+        jesterTest.TellJoke();
+
+        // Assert with moq
+        moqJokeOutput.Verify(x => x.OutputJoke(testJoke));
+    }
+
+
+    [Fact]
+    public void OuputJoke_ChuckNorrisJoke_SuccessfulSkip()
+    {
+        // Arrange
+        var moqJokeService = new Mock<IJokeService>();
+        var moqJokeOutput = new Mock<IJokeOutput>();
+        string chuckJoke = "Chuck Norris";
+        string testJoke = "test joke";
+        moqJokeService.SetupSequence(x => x.GetJoke()).Returns(chuckJoke).Returns(testJoke);
+        Jester jesterTest = new(moqJokeService.Object, moqJokeOutput.Object);
+
+        // Act 
+        jesterTest.TellJoke();
+
+        // Assert with moq
+        // Should skip chuckJoke and equal testJoke
+        moqJokeOutput.Verify(x => x.OutputJoke(testJoke));

```

Comment:
This actually doesn't check that chuckJoke wasn't run.

You want something like:
```suggestion
        moqJokeOutput.Verify(x => x.OutputJoke(It.Any<string>()), Times.Once);

```
This makes sure for any input to OutputJoke that it was only called one. You just made sure that OutputJoke was called at all with the output of testJoke.